fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/lock/InMemoryResourceLockManager.java

```diff
@@ -60,11 +72,29 @@ public void acquire(final String txId, final FedoraId resourceId) {
                         String.format("Cannot update %s because it is being updated by another transaction.",
                                 resourceIdStr));
             }
+            final String estimateParentPath = resourceIdStr.substring(0, resourceIdStr.lastIndexOf('/'));
```

Comment:
In the case of the root resource, `resourceIdStr.lastIndexOf('/')` will return `-1`